python/setup.py

```diff
@@ -1,13 +1,17 @@
 # Copyright (c) 2019-2022, NVIDIA CORPORATION.
 
+import os
+
 from setuptools import find_packages
 from skbuild import setup
 
 import versioneer
 
 setup(
-    name="rmm",
-    version=versioneer.get_version(),
+    name="rmm" + os.getenv("RAPIDS_PY_WHEEL_CUDA_SUFFIX", default=""),
+    version=os.getenv(
+        "RAPIDS_PY_WHEEL_VERSIONEER_OVERRIDE", default=versioneer.get_version()
```

Comment:
When is this override needed?

Reply:
We use this override to build every pip wheel:
* For nightly wheels, the input override parameter is left blank, which results in the override being computed: https://github.com/rapidsai/shared-action-workflows/blob/main/cibuildwheel/action.yml#L90
    * You'll notice it uses `rapids-pip-wheel-version $(epoch-timestamp)`: https://github.com/sevagh/gha-tools/blob/feat/pip-wheel-version-script/tools/rapids-pip-wheel-version
    * This does: git tag v22.12.00a -> strip letters 22.12.00 -> distutils normalize -> 22.12.0 -> append epoch timestamp 22.12.0.165846854...
* For stable wheels, the input override parameter is set by the person calling the workflow, i.e. they would use `22.12.0` to create a wheel that has a stable version for the 22.12.0 release

Reply:
I'm confused by the nightly and stable naming. So the nightly doesn't have an `a` or `pre` or similar tag to indicate it's a prerelease? Are the resulting versions ordered such that nightlies aren't considered "newer" than the actual release? I would expect that `22.12.0.1668097661` (the nightly name) would be considered a higher version than the stable `22.12.0` (not yet released). Is that correct? Ideally the stable release would be considered greater than the nightlies preceding it.

Reply:
Yes, you're correct. The elephant in the room with `--pre` is that using `pip install --pre` will spread to every dependency; pre-release versions of numpy, scikit, every dependency.

So, that's why Vyas and I invented a "stable-looking" (but still nightly) versioning scheme.

On release day, what will happen is:

In the dev/nightly index, indeed:
22.12.0 (stable) < 22.12.0.$timestamp1 < 22.12.0.$timestamp2 < ... < 22.12.0.$latest_timestamp

In the stable NGC index, the only wheel present will be:
22.12.0 (stable)

So:
1. `pip install rmm-cu11 --extra-index-url=<stable-index>` will only install the latest stable release: 22.10.0 currently, and 22.12.0 in the next release (due to only stable releases existing in this index)
2. `pip install rmm-cu11 --extra-index-url=<nightly-index>` will install the latest nightly timestamp, 22.12.0.16384384
3. `pip install rmm-cu11==22.12.0 --extra-index-url=<stable-or-nightly>` will do the right thing when pointed at both stable or production

Finally, in the release scripts, what I generally do is:
```
pip download rmm-cu11==22.12.0 --index-url=<dev-index> # get latest stable wheel, avoiding epoch-timestamped nightlies

<promote rmm_cu11-22.12.0-cp38-cp38-manylinux...whl to the stable index>
```

So, using `==22.12.0` as a strict version specifier will ensure we avoid the timestamped nightlies.

Reply:
I would prefer a version scheme where upgrades across index URLs work in a normal way. I think `--pre` may be a feature, not a bug. Requiring the pinning `==22.12.0` to get the stable release breaks the package upgrade path.

Reply:
Discussed this offline with Bradley. Getting the nightly and stable index installs to coexist by leveraging pre-releases becomes extremely difficult because non-RAPIDS dependencies will upload pre-releases to PyPI and `--pre` is a blanket setting that would affect both RAPIDS installs (from our index) and the installations from PyPI. For now, we're pushing the responsibility for managing stable vs nightly installs of RAPIDS libraries to the user. If we are able to devise a superior version scheme in the future it is definitely worth revisiting.